jobs/base_company_sample.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import string
+
+# [START instantiate]
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+client_service = build('jobs', 'v2')
+
+# [END instantiate]
+
+
+# [START basic_company]
+def generate_company():
+    # distributor company id should be a unique Id in your system.
+    distributor_company_id = 'company:' + ''.join(
+        random.choice(string.ascii_uppercase + string.digits)
+        for _ in range(16))
+
+    display_name = 'Google'
+    hq_location = '1600 Amphitheatre Parkway Mountain View, CA 94043'
+
+    company = {
+        'display_name': display_name,
+        'distributor_company_id': distributor_company_id,
+        'hq_location': hq_location
+    }
+    print('==========\nCompany generated: %s\n==========' % company)
+    return company
+
+
+# [END basic_company]
+
+
+# [START create_company]
+def create_company(client_service, company_to_be_created):
+    try:
+        company_created = client_service.companies().create(
+            body=company_to_be_created).execute()
+        print('==========\nCompany created: %s\n==========' % company_created)
+        return company_created
+    except HttpError as e:
```

Comment:
Use the base error class would be better I think.

Reply:
Done.